packages/sdk/identity/src/odis/identifier.ts

```diff
@@ -0,0 +1,202 @@
+    blsBlindingClient = new WasmBlsBlindingClient(context.odisPubKey)
+  }
+
+  const base64BlindedMessage = await getBlindedIdentifier(
+    offchainIdentifier,
+    blsBlindingClient,
+    seed
+  )
+
+  const base64BlindSig = await getBlindedIdentifierSignature(
+    account,
+    signer,
+    context,
+    base64BlindedMessage,
+    clientVersion,
+    sessionID
+  )
+
+  return getOnchainIdentifierFromSignature(
+    offchainIdentifier,
+    identifierType,
+    base64BlindSig,
+    blsBlindingClient
+  )
+}
+
+/**
+ * Blinds the off-chain identifier in preparation for the ODIS request
+ * Caller should use the same blsBlindingClient instance for unblinding
+ */
+export async function getBlindedIdentifier(
+  identifier: string,
+  blsBlindingClient: BlsBlindingClient,
+  seed?: Buffer
+): Promise<string> {
+  debug('Retrieving blinded message')
+  const base64Identifier = Buffer.from(identifier).toString('base64')
+  return blsBlindingClient.blindMessage(base64Identifier, seed)
+}
+
+/**
+ * Query ODIS for the blinded signature
+ * Response can be passed into getIdentifierIdentifierFromSignature
+ * to retrieve the on-chain identifier
+ */
+export async function getBlindedIdentifierSignature(
+  account: string,
+  signer: AuthSigner,
+  context: ServiceContext,
+  base64BlindedMessage: string,
+  clientVersion?: string,
+  sessionID?: string,
+  keyVersion?: number
+): Promise<string> {
+  const body: SignMessageRequest = {
+    account,
+    blindedQueryPhoneNumber: base64BlindedMessage,
+    version: clientVersion,
+    authenticationMethod: signer.authenticationMethod,
+    sessionID,
+  }
+
+  const response = await queryOdis(
+    body,
+    context,
+    CombinerEndpoint.PNP_SIGN,
+    SignMessageResponseSchema,
+    {
+      [KEY_VERSION_HEADER]: keyVersion?.toString(),
+      Authorization: await getOdisPnpRequestAuth(body, signer),
+    }
+  )
+
+  if (!response.success) {
+    throw new Error(response.error)
+  }
+
+  return response.signature
+}
+
+/**
+ * Unblind the response and return the on-chain identifier
+ */
+export async function getOnchainIdentifierFromSignature(
+  offchainIdentifier: string,
+  identifierType: string | IdentifierType,
+  base64BlindedSignature: string,
+  blsBlindingClient: BlsBlindingClient
+): Promise<IdentifierHashDetails> {
+  debug('Retrieving unblinded signature')
+  const base64UnblindedSig = await blsBlindingClient.unblindAndVerifyMessage(base64BlindedSignature)
+  const sigBuf = Buffer.from(base64UnblindedSig, 'base64')
+
+  debug('Converting sig to pepper')
+  const pepper = getPepperFromThresholdSignature(sigBuf)
+  const identifierHash = getIdentifierHash(sha3, offchainIdentifier, identifierType, pepper)
+  return { offchainIdentifier, identifierHash, pepper, unblindedSignature: base64UnblindedSig }
+}
+
+export const getIdentifierHash = (
+  sha3: (a: string) => string | null,
+  offchainIdentifier: string,
+  identifierType: string | IdentifierType,
+  pepper?: string
```

Comment:
this variable used to be a salt in the old code, but it's a little confusing that we suddenly switch to salt when we've been referring to it as a pepper the whole time